rust/agama-software/src/callbacks/commit_download.rs

```diff
@@ -0,0 +1,60 @@
+use agama_utils::{actor::Handler, api::question::QuestionSpec, progress, question};
+use tokio::runtime::Handle;
+use zypp_agama::callbacks::PkgDownloadCallbacks;
+
+struct CommitDownload {
+    progress: Handler<progress::Service>,
+    questions: Handler<question::Service>,
+}
+
+impl CommitDownload {
+    fn new(progress: Handler<progress::Service>, questions: Handler<question::Service>) -> Self {
+        Self {
+            progress,
+            questions,
+        }
+    }
+}
+
+impl PkgDownloadCallbacks for CommitDownload {
+    fn start_preload(&self) {
+        // TODO: report progress that we start preloading packages
+        tracing::info!("Start preload");
+    }
+
+    fn problem(
+        &self,
+        name: &str,
+        error: zypp_agama::callbacks::DownloadResolvableError,
+        description: &str,
+    ) -> zypp_agama::callbacks::ProblemResponse {
+        // TODO: make it generic for any problem questions
+        // TODO: localization
```

Comment:
OK, the localization part is being held back by me,
but we could be marking up translatable texts using the crate that we already are using:
```rs
use gettextrs::gettext;
...
        let actions = [("Retry", gettext("Retry")), ("Ignore", gettext("Ignore"))];
```